Makefile

```diff
@@ -0,0 +1,173 @@
+#         pre-fetched but evaluated at time of use.
+#
+
+VERSION := $(shell (cd "$(SRCDIR)" && grep "^Version:" image-builder-cli.spec | sed 's/[^[:digit:]]*\([[:digit:]]\+\).*/\1/'))
+COMMIT = $(shell (cd "$(SRCDIR)" && git rev-parse HEAD))
+
+#
+# Generic Targets
+#
+# The following is a set of generic targets used across the makefile. The
+# following targets are defined:
+#
+#     help
+#         This target prints all supported targets. It is meant as
+#         documentation of targets we support and might use outside of this
+#         repository.
+#         This is also the default target.
+#
+#     $(BUILDDIR)/
+#     $(BUILDDIR)/%/
+#         This target simply creates the specified directory. It is limited to
+#         the build-dir as a safety measure. Note that this requires you to use
+#         a trailing slash after the directory to not mix it up with regular
+#         files. Lastly, you mostly want this as order-only dependency, since
+#         timestamps on directories do not affect their content.
+#
+
+.PHONY: help
+help:
+	@echo "make [TARGETS...]"
+	@echo
+	@echo "This is the maintenance makefile of image-builder-cli. The following"
+	@echo "targets are available:"
+	@echo
+	@echo "    help:               Print this usage information."
+	@echo "    rpm:                Build the RPM"
+	@echo "    srpm:               Build the source RPM"
+	@echo "    scratch:            Quick scratch build of RPM"
+	@echo "    clean:              Remove all built binaries"
+
+$(BUILDDIR)/:
+	mkdir -p "$@"
+
+$(BUILDDIR)/%/:
+	mkdir -p "$@"
+
+
+#
+# Maintenance Targets
+#
+# The following targets are meant for development and repository maintenance.
+# They are not supported nor is their use recommended in scripts.
+#
+
+.PHONY: build
+build: $(BUILDDIR)/bin/
+	go build -o $<image-builder ./cmd/image-builder/
+
+.PHONY: clean
+clean:
+	rm -rf $(BUILDDIR)/bin/
+	rm -rf $(CURDIR)/rpmbuild
+	rm -rf $(CURDIR)/release_artifacts
+
+#
+# Building packages
+#
+# The following rules build image-builder-cli packages from the current HEAD
+# commit, based on the spec file in this directory.  The resulting packages
+# have the commit hash in their version, so that they don't get overwritten
+# when calling `make rpm` again after switching to another branch.
+#
+# All resulting files (spec files, source rpms, rpms) are written into
+# ./rpmbuild, using rpmbuild's usual directory structure.
+#
+
+RPM_SPECFILE=rpmbuild/SPECS/image-builder-cli.spec
+RPM_TARBALL=rpmbuild/SOURCES/image-builder-cli-$(COMMIT).tar.gz
+
+.PHONY: $(RPM_SPECFILE)
+$(RPM_SPECFILE):
+	mkdir -p $(CURDIR)/rpmbuild/SPECS
+	git show HEAD:image-builder-cli.spec > $(RPM_SPECFILE)
+	./tools/rpm_spec_add_provides_bundle.sh $(RPM_SPECFILE)
+
+RPM_TARBALL_UNCOMPRESSED=$(RPM_TARBALL:.tar.gz=.tar)
+
+$(RPM_TARBALL):
+	mkdir -p $(CURDIR)/rpmbuild/SOURCES
+	git archive --prefix=image-builder-cli-$(COMMIT)/ --format=tar.gz HEAD > $(RPM_TARBALL)
```

Comment:
This should probably consider release tags `v*` so that our release tarballs have a nice filename that consists only of the name and version but no commit ref.

Reply:
@mvo5 @supakeen @thozza should I just manually set the tag `v1` for now?